### Main imports

In [6]:
import numpy as np
import pandas as pd
from sklearn.svm import SVR
from sklearn.linear_model import Lasso
from sklearn.metrics import mean_squared_error
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split

### Regression sample

In [37]:
# input data reading
df = pd.read_csv("./winequality-red.csv", sep=";")
df.head()

fixed acidity,volatile acidity,citric acid,residual sugar,chlorides,free sulfur dioxide,total sulfur dioxide,density,pH,sulphates,alcohol,quality
0   7.4,0.7,0,1.9,0.076,11,34,0.9978,3.51,0.56,9.4,5                                                                                               
1   7.8,0.88,0,2.6,0.098,25,67,0.9968,3.2,0.68,9.8,5                                                                                               
2  7.8,0.76,0.04,2.3,0.092,15,54,0.997,3.26,0.65,...                                                                                               
3  11.2,0.28,0.56,1.9,0.075,17,60,0.998,3.16,0.58...                                                                                               
4   7.4,0.7,0,1.9,0.076,11,34,0.9978,3.51,0.56,9.4,5

In [62]:
# independent variables and dependent variable
x = df.loc[:, : "alcohol"]
# y = df["quality"]
y = df.loc[: , "quality" :]

In [70]:
# features have to be scaled before model learning
scaler = StandardScaler()
x = scaler.fit_transform(x)

ValueError: at least one array or dtype is required

In [64]:
# spliting data to train and test parts
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=1)

In [65]:
# create and fit LASSO and SVR models 
lasso = Lasso(alpha=0.001)
svr = SVR(C=8, epsilon=0.2, gamma=0.5)
lasso.fit(x_train, y_train)
svr.fit(x_train, y_train)

ValueError: at least one array or dtype is required

In [66]:
# make prediction on test data and rounding results
y_pred_lasso = np.round(np.clip(lasso.predict(x_test), 1, 10)).astype(int)
y_pred_svr = np.round(np.clip(svr.predict(x_test), 1, 10)).astype(int)

NotFittedError: This Lasso instance is not fitted yet. Call 'fit' with appropriate arguments before using this estimator.

In [67]:
np.round(1 - mean_squared_error(y_test, y_pred_lasso) / y_test.std(), 2)

NameError: name 'y_pred_lasso' is not defined

In [68]:
np.round(1 - mean_squared_error(y_test, y_pred_svr) / y_test.std(), 2)

NameError: name 'y_pred_svr' is not defined

This LASSO model describes just 28% of initial data dispersion. With same data SVR describes 50%. Here SVR build nonlinear decision boundary using so-called "kernel trick"